libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt

```diff
@@ -149,13 +148,11 @@ class RustMatrixClient constructor(
 
     init {
         client.setDelegate(clientDelegate)
-        rustSyncService.syncState
-            .onEach { syncState ->
-                if (syncState == SyncState.Running) {
-                    onSlidingSyncUpdate()
-                }
+        roomListService.state.onEach { state ->
+            if (state == RoomListService.State.Running) {
+                onSlidingSyncUpdate()
```

Comment:
Maybe rename this private fun, since it's not called anymore when the sync service is updated (and it was already not really the case: the existing name was not ideal). I suggest to rename to `onRoomListServiceRunning`.

It was confusing me during the review, but I see that this fun only get the verification controller.

```kotlin
    private fun onSlidingSyncUpdate() {
        if (!verificationService.isReady.value) {
            try {
                verificationService.verificationController = client.getSessionVerificationController()
            } catch (e: Throwable) {
                Timber.e(e, "Could not start verification service. Will try again on the next sliding sync update.")
            }
        }
    }
```

Also for clarity, I would replace 
` if (!verificationService.isReady.value) {`
 by
` if (verificationService.verificationController == null) {`
I think this is equivalent.

Maybe update the Timber log too.